### ISSUE TITLE
Quirky pyright warnings

### ISSUE THREAD

**Issue by DragonRulerX** [microsoft/pyright#7352]
**Describe the bug**
I recently had my code linted by pyright after I verified everything was working and it found a few things I'd consider questionable.
I'm not sure if these should have been reported as separate bugs or not since I'm not sure if any of them are "bugs" or simply differing from personal preference/expectations so I figured I'd post them all here and let the experts decide.

<hr />

1. This might be a quirk of me using PyInstaller, but the following code yielded the following error:
> "_MEIPASS" is not a known member of module "sys" Pyright(reportAttributeAccessIssue)
```py
import sys
...
if getattr(sys, "frozen", False):
    _ASSETS_DIR = Path(sys._MEIPASS) / "assets"
else:
    _ASSETS_DIR = _BASEDIR / "assets"
```
I also tried:
```py
import sys
...
if hasattr(sys, "_MEIPASS"):
    _ASSETS_DIR = Path(sys._MEIPASS) / "assets"
else:
    _ASSETS_DIR = _BASEDIR / "assets"
```
Ref:
- https://pyinstaller.org/en/v3.2.1/runtime-information.html#using-file-and-sys-meipass

The only way to clear this was by using this:
```py
import sys
...
if hasattr(sys, "_MEIPASS"):
    _ASSETS_DIR = Path(getattr(sys, "_MEIPASS")) / "assets"
else:
    _ASSETS_DIR = _BASEDIR / "assets"
```
I would have assumed that at least the `hasattr` check variation should have given pyright enough context to accept writing `sys._MEIPASS`, but it did not sadly.

<hr />

2. Attempting to set an attribute on a function object is apparently frowned upon despite it working just fine
> Cannot assign member "ran" for type "function"
>    Member "ran" is unknown Pyright(reportFunctionMemberAccess)
```py
def init():
    init.ran = True
```
However, this was accepted
```py
def init():
    setattr(init, "ran", True)
```

<hr />

3. I'm using the `tkinter` library and apparently the "bind" method gets mad when you use a natural function for the 2nd parameter.
> No overloads for "bind" match the provided arguments Pyright(reportCallIssue)
> %LOCALAPPDATA%\Sublime Text\Package Storage\LSP-pyright\20.11.0\language-server\node_modules\pyright\dist\typeshed-fallback\stdlib\tkinter\\_\_init\_\_.pyi:447
> Overload 2 is the closest match
> Argument of type "() -> None" cannot be assigned to parameter "func" of type "str" in function "bind"
> "function" is incompatible with "str" Pyright(reportArgumentType)
```py
import tk

app = tk.Tk()
app.bind("<Escape>", close)

def close():
    app.destroy()

app.mainloop()
```

**VS Code extension or command-line**
Sublime Text 4 Build 4169

**Comment by erictraut** [comment]
As I mentioned in another thread, pyright isn't a linter. It's a type checker.

The type information for stdlib libraries comes from the [typeshed project](https://github.com/python/typeshed). All major Python type checkers use these same stub files. If you think that some of the type information in these stub files is incorrect, please file a bug in the typeshed project. Pyright bundles the latest typeshed stubs with it. This applies to point 3 and probably point 1 in your post. I can't tell for sure with point 1 because the code samples you've posted above are incomplete. 

For point 2, pyright is working as intended. You can disable the `reportFunctionMemberAccess` diagnostic check if you don't want it to report these diagnostics.

**Comment by DragonRulerX** [comment]
Ah, I forgot to include the `pathlib` import, my bad - edited the above post.
Also, I poked at point 3 more and found a problem with my code - I'm not super familiar with `tkinter` yet.
I do think point 1 holds though and is likely due to the dynamic nature of things that can end up in the `sys` module.
Perhaps you disagree?
